src/data/number/numberFn.ts

```diff
@@ -1,10 +1,90 @@
 import { assert, assertInteger } from '../../utils/assert'
-import { plString } from '../string/stringFn'
 import { PLNumber } from './PLNumber'
 
-export const plNumber = (value: number): PLNumber => new PLNumber(value)
+export interface DecimalResult {
+  intValue: number
+  decimals: number
+}
+
+export function plFloat(x: number): PLNumber {
+  return parseNumber(x.toString())
+}
+
+export function plNumber(intValue: number, decimals = 0): PLNumber {
+  assertInteger(intValue)
+  assertInteger(decimals)
+  return new PLNumber(intValue, decimals)
+}
+
+export function parseNumber(strValue: string): PLNumber {
+  assertNumeric(strValue)
+  const decimalObj = isScientific(strValue) ? parseScientificString(strValue) : parseDecimalString(strValue)
+  return plNumber(decimalObj.intValue, decimalObj.decimals)
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertNumeric(strValue: any): boolean {
```

Comment:
please do not use `any` type, espeacially when the parameter name suggests `string` type

Reply:
See my previous comment on the same issue [here](https://github.com/maxinteger/pocket-lisp-stdlib/pull/14#discussion_r559078636).

Reply:
Sorry but your comment still not explain why do you use `any` type for `strValue`
Either the parameters name is wrong or the type of the parameter so you should
- use `string` type: `assertNumeric(strValue: string): boolean` OR
- use `unknown` type with proper parameter name: `assertNumeric(value: unknown): boolean`

Reply:
The problem is that `isNan` expects a number (even though I want to apply it to a string) so both `string` and `unknown` will give an error:

```
Argument of type 'string' is not assignable to parameter of type 'number'.
Argument of type 'unknown' is not assignable to parameter of type 'number'.
```

Reply:
I see, you can do this then:

```
export function assertNumeric(strValue: string): boolean {
   return assert(strValue === '' || isNaN(Number(strValue)), `Invalid number: "${strValue}"`)
}
```

Reply:
Thanks! It works now.